NoSuchMethodError v8.0.1 Android 10-12

Hi, just updated to v8.0.1 and started getting below crash reports.
Both java.nio.charset.StandardCharsets and java.net.URLEncoder shoulsd be avalible on Android 10 and 12. 
Proguard/R8 cannot be the reason as it  works on our test devices  with Android 14,15,16

Sample 1 - Android 10:
`java.lang.NoSuchMethodError: No static method encode(Ljava/lang/String;Ljava/nio/charset/Charset;)Ljava/lang/String; in class Ljava/net/URLEncoder; or its super classes (declaration of 'java.net.URLEncoder' appears in /apex/com.android.runtime/javalib/core-oj.jar)
at com.dropbox.core.DbxRequestUtil.encodeUrlParam(DbxRequestUtil.java:3)
at com.dropbox.core.DbxRequestUtil.encodeUrlParams(DbxRequestUtil.java:15)
at com.dropbox.core.DbxRequestUtil.buildUrlWithParams(DbxRequestUtil.java:5)
at com.dropbox.core.android.AuthActivity.startWebAuth(AuthActivity.kt:177)
at com.dropbox.core.android.AuthActivity.onTopResumedActivityChanged$lambda$0(AuthActivity.kt:29)
at com.dropbox.core.android.AuthActivity.a(AuthActivity.kt:1)
at y31.run(R8$$SyntheticClass:7)
at android.app.Activity.runOnUiThread(Activity.java:6924)
at com.dropbox.core.android.AuthActivity.onTopResumedActivityChanged(AuthActivity.kt:130)`

Sample 2 - Android 12:
`java.lang.RuntimeException: An error occurred while executing doInBackground()
at android.os.AsyncTask$4.done(AsyncTask.java:415)
at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
at java.util.concurrent.FutureTask.run(FutureTask.java:271)
at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:305)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
at java.lang.Thread.run(Thread.java:920)
Caused by: java.lang.NoSuchMethodError: No static method encode(Ljava/lang/String;Ljava/nio/charset/Charset;)Ljava/lang/String; in class Ljava/net/URLEncoder; or its super classes (declaration of 'java.net.URLEncoder' appears in /apex/com.android.art/javalib/core-oj.jar)
at com.dropbox.core.DbxRequestUtil.encodeUrlParam(DbxRequestUtil.java:3)
at com.dropbox.core.DbxRequestUtil.encodeUrlParams(DbxRequestUtil.java:51)
at com.dropbox.core.DbxRequestUtil.startPostNoAuth(DbxRequestUtil.java:5)
at com.dropbox.core.DbxRequestUtil$2.run(DbxRequestUtil.java:13)
at com.dropbox.core.DbxRequestUtil.runAndRetry(DbxRequestUtil.java:4)
at com.dropbox.core.DbxRequestUtil.doPostNoAuth(DbxRequestUtil.java:17)
at com.dropbox.core.DbxPKCEManager.makeTokenRequest(DbxPKCEManager.java:65)
at com.dropbox.core.android.internal.TokenRequestAsyncTask.doInBackground(TokenRequestAsyncTask.kt:15)
at com.dropbox.core.android.internal.TokenRequestAsyncTask.doInBackground(TokenRequestAsyncTask.kt:47)`

--- a/core/src/main/java/com/dropbox/core/DbxOAuth1Upgrader.java
+++ b/core/src/main/java/com/dropbox/core/DbxOAuth1Upgrader.java
@@ -10,8 +10,8 @@ import com.fasterxml.jackson.core.JsonParser;
 import com.fasterxml.jackson.core.JsonToken;
 
 import java.io.IOException;
+import java.io.UnsupportedEncodingException;
 import java.net.URLEncoder;
-import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 
 /**
@@ -105,7 +105,14 @@ public final class DbxOAuth1Upgrader
 
     private static String encode(String s)
     {
-        return URLEncoder.encode(s, StandardCharsets.UTF_8);
+        try {
+            // Use the String overload for Android API compatibility. The Charset overload was
+            // added to Android much later than the String overload.
+            return URLEncoder.encode(s, "UTF-8");
+        } catch (UnsupportedEncodingException ex) {
+            // UTF-8 is required by every Java implementation.
+            throw new AssertionError(ex);
+        }
     }
 
     /**

--- a/core/src/main/java/com/dropbox/core/DbxRequestUtil.java
+++ b/core/src/main/java/com/dropbox/core/DbxRequestUtil.java
@@ -1,11 +1,11 @@
 package com.dropbox.core;
 
 import java.io.IOException;
+import java.io.UnsupportedEncodingException;
 import java.net.URI;
 import java.net.URISyntaxException;
 import java.net.URLEncoder;
 import java.nio.charset.CharacterCodingException;
-import java.nio.charset.StandardCharsets;
 import java.util.concurrent.TimeUnit;
 import java.util.ArrayList;
 import java.util.List;
@@ -38,7 +38,14 @@ public final class DbxRequestUtil {
     public static DbxGlobalCallbackFactory sharedCallbackFactory;
 
     public static String encodeUrlParam(String s) {
-        return URLEncoder.encode(s, StandardCharsets.UTF_8);
+        try {
+            // Use the String overload for Android API compatibility. The Charset overload was
+            // added to Android much later than the String overload.
+            return URLEncoder.encode(s, "UTF-8");
+        } catch (UnsupportedEncodingException ex) {
+            // UTF-8 is required by every Java implementation.
+            throw new AssertionError(ex);
+        }
     }
 
     public static String buildUrlWithParams(/*@Nullable*/String userLocale,

--- a/core/src/test/java/com/dropbox/core/DbxRequestUtilTest.java
+++ b/core/src/test/java/com/dropbox/core/DbxRequestUtilTest.java
@@ -1,0 +1,14 @@
+package com.dropbox.core;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import org.testng.annotations.Test;
+
+public class DbxRequestUtilTest {
+
+    @Test
+    public void testEncodeUrlParam() {
+        assertThat(DbxRequestUtil.encodeUrlParam("hello world/+"))
+            .isEqualTo("hello+world%2F%2B");
+    }
+}

